test/inductor/test_triton_kernels.py

```diff
@@ -2070,6 +2070,47 @@ def fn(x):
         res2 = fn_c(x2)
         self.assertEqual(x2 * x2, res2)
 
+    @requires_gpu
+    def test_triton_kernel_none_args(self):
+        # https://github.com/pytorch/pytorch/issues/115344
+        @triton.autotune(  # E: Untyped decorator makes function "sin_kernel" untyped  [misc]
+            configs=[
+                triton.Config({"BLOCK_SIZE": 32}, num_stages=5, num_warps=2),
+                triton.Config({"BLOCK_SIZE": 64}, num_stages=4, num_warps=4),
+            ],
+            key=["n_elements"],
+        )
+        @triton.jit  # E: Untyped decorator makes function "sin_kernel" untyped  [misc]
+        def sin_kernel(  # E: Function is missing a return type annotation  [no-untyped-def]
+            in_ptr0,
+            out_ptr,
+            n_elements,
+            BLOCK_SIZE: "tl.constexpr",
+        ):
+            pid = tl.program_id(axis=0)
+            block_start = pid * BLOCK_SIZE
+            offsets = block_start + tl.arange(0, BLOCK_SIZE)
+            mask = offsets < n_elements
+            if in_ptr0 is not None:
+                x = tl.load(in_ptr0 + offsets, mask=mask)
+            else:
+                x = 0.0
+            output = tl.sin(x)
+            tl.store(out_ptr + offsets, output, mask=mask)
+
+        def sin_triton(x, out):
+            n_elements = out.numel()
+            sin_kernel[(n_elements,)](x, out, n_elements)
+
+        x = torch.randn(65, device="cuda")
+        out = torch.empty_like(x)
+        out_compiled = torch.empty_like(x)
+        sin_triton_compiled = torch.compile(fullgraph=True)(sin_triton)
+        for first in (x, None):
```

Comment:
Would it make sense to unroll the loop, so that if the test fails we know whether it was with `x` or `None`? Also, I'd suggest doing `None` first, as it's more likely to have issues after `None` than before, I believe.